website/api/zkvm/tutorials/io.md

```diff
@@ -0,0 +1,329 @@
+[`init` guest][example-voting-machine-guest-elf-init] program and immediately
+[`commit`][example-voting-machine-guest-elf-init-commit]ed to the journal.
+Note how easy it is to operate on data structures when using the standard
+[`read`][source-env::read] and [`commit`][source-env::commit] functions, no bit
+manipulation needed. It'd be a different story if we were using the `_slice`
+variants. Since we don't have to worry about performance critical code here, we
+can safely use the standard functions.
+
+Moving on to the [`submit`][example-voting-machine-host-submit] function, we can
+see that in the host an [`output`
+buffer][example-voting-machine-host-submit-output-buffer] is passed to the
+[`stdout`][example-voting-machine-host-submit-output-stdout] file descriptor of
+the guest.
+It'll be filled with values produced by the guest and then read by calling the
+[`from_slice`][source-from_slice] method on the buffer.
+This can be seen in [this
+line][example-voting-machine-host-submit-output-from_slice].
+The result that was filled in the buffer came from the
+[`write`][example-voting-machine-guest-elf-submit-write] method call in the
+guest.
+Remember, the [`write`][source-env::write] method is used to send data to the
+host's [`stdout`][stdout] file descriptor.
+
+Still in the [`submit`][example-voting-machine-host-submit] function, note how
+the _private_ output from the guest is used, and how it's relevant the
+distinction between _public_ and _private_ data in this case.
+In the example presented, the `VotingMachineState` struct is changed during the
+guest's execution. But we don't want to commit (make public) the state of the
+voting machine, so we use the `stdout` file descriptor to send the result back
+to the host.
+This way, we can update the voting machine state at each iteration while
+preserving its privacy.
+
+Finally, in the [`freeze`][example-voting-machine-host-freeze] function, the
+same patterns of sending and receiving data are repeated.
+
+## Conclusions
+
+In this guide, we've covered the basics of I/O in the zkVM. We've seen how to
+send data from the host to the guest and vice-versa, how _private_ and _public_
+data are distinguished, and how to commit data to the journal.
+We also covered the trade-offs between using the standard functions and their
+`_slice` variants and showed through the [Voting Machine
+example][example-voting-machine] how to implement I/O in practice.
+There are more examples available in the [examples page] that you can use as
+reference if you wish.
+
+Happy coding!
+
+[article-composition]: https://www.risczero.com/blog/proof-composition
+[docs-guest-code-optmization]: ../optimization.md#when-reading-data-as-raw-bytes-use-envread_slice
+[docs-hello-world]: ./hello-world.md
+[docs-remote-proving]: ../../generating-proofs/remote-proving.md
+[example-chess-core]: https://github.com/risc0/risc0/blob/main/examples/chess/core/src/lib.rs
```

Comment:
I think our best practice is to point to the latest release rather than `main`, although looking around at other files, it looks like we point to `main` more often than I had thought...
```suggestion
[example-chess-core]: https://github.com/risc0/risc0/blob/v0.20.1/examples/chess/core/src/lib.rs
```

Reply:
If it's in next, we point to main, if it's a specific version, part of the promotion process is to go thru each main link and point it to the appropriate stable link.